user/pages/03.commerce2/02.developer-guide/06.stores/01.resolve-current-store/docs.md

```diff
@@ -4,4 +4,73 @@ taxonomy:
     category: docs
 ---
 
+The first store created in Commerce is saved as the default store in commerce_store.settings.yml.
+
+This uses Drupal\commerce_store\Resolver\DefaultStoreResolver to retrieve it on page load.
+
+In order to load a different store, create a new StoreResolver.
+
+In this example, use a store ID set in a cookie to load the relevant store.
+
+Copy DefaultStoreResolver to your module (in /src/Resolver/) and rename to CookieStoreResolver.
+Change the code in resolve() to load the store ID from the cookie and return it.
+
+```php
+<?php
+
+namespace Drupal\cookie_store\Resolver;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\commerce_store\Resolver\StoreResolverInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Drupal\commerce_store\Entity\Store;
+
+/**
+ * Returns the store for an ID set in a cookie.
+ */
+class CookieStoreResolver implements StoreResolverInterface {
+
+  /**
+   * The store storage.
+   *
+   * @var \Drupal\commerce_store\StoreStorageInterface
+   */
+  protected $storage;
+
+  /**
+   * Constructs a new CookieStoreResolver object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->storage = $entity_type_manager->getStorage('commerce_store');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function resolve() {
+      $store_id = Request::createFromGlobals()->cookies->get('Drupal_visitor_store_id');
```

Comment:
This should be either `$this->requestStack->getCurrentRequest()` or in constructor set current request that way and `$this->request->cookies->get(..)`